akka-persistence/src/main/scala/akka/persistence/PersistentActor.scala

```diff
@@ -562,6 +606,28 @@ abstract class AbstractPersistentActor extends AbstractActor with AbstractPersis
   def deferAsync[A](event: A)(handler: Procedure[A]): Unit =
     internalDeferAsync(event)(event ⇒ handler(event))
 
+  /**
+   * Defer the handler execution until all pending handlers have been executed. It is guaranteed that no new commands
+   * will be received by a persistent actor between a call to `defer` and the execution of its `handler`.
+   * Allows to define logic within the actor, which will respect the invocation-order-guarantee
+   * in respect to `persistAsync` or `persist` calls. That is, if `persistAsync` or `persist` was invoked before `defer`,
+   * the corresponding handlers will be invoked in the same order as they were registered in.
+   *
+   * This call will NOT result in `event` being persisted, use `persist` or `persistAsync` instead
+   * if the given event should possible to replay.
+   *
+   * If there are no pending persist handler calls, the handler will be called immediately.
+   *
+   * If persistence of an earlier event fails, the persistent actor will stop, and the `handler`
+   * will not be run.
+   *
+   * @param event event to be handled in the future, when preceding persist operations have been processes
+   * @param handler handler for the given `event`
+   */
+  def defer[A](event: A)(handler: A ⇒ Unit): Unit = {
```

Comment:
handler should be `Procedure` in Java API

Reply:
Fixed.